samples/rpi/blinky.cpp

```diff
@@ -0,0 +1,132 @@
+      },
+      // On Exit Action
+      []() {},
+      // Receivers
+      std::map<const std::string, Receiver>{
+          {"started",
+           [&](const Address &sender, std::istream &args) {
+             std::string child;
+             args >> child;
+             info() << "Parent: " << sender << ' ' << child << " has started\n"
+                    << std::flush;
+             addresses.insert(child, sender);
+           }},
+          {"pulsed",
+           [&](const Address &sender, std::istream &args) {
+             std::string child;
+             args >> child;
+             info() << "Parent: " << sender << ' ' << child << " has pulsed\n"
+                    << std::flush;
+           }},
+          {"errored",
+           [&](const Address &sender, std::istream &args) {
+             std::string child;
+             args >> child;
+             std::ostringstream os;
+             os << args.rdbuf();
+             info() << "Parent: " << sender << ' ' << child
+                    << " has errored: " << os.str() << '\n'
+                    << std::flush;
+           }},
+          {"exited",
+           [&](const Address &sender, std::istream &args) {
+             std::string child;
+             args >> child;
+             info() << "Parent: " << sender << ' ' << child << " has exited\n"
+                    << std::flush;
+             addresses.remove(child);
+             m.Spawn(child); // Retry
+           }},
+      }));
+
+  m.AddState(std::make_unique<State>(
+      // State Name
+      "loop",
+      // Parent State
+      "main",
+      // On Entry Action
+      []() {
+        // Request Button state
+        if (const auto a = addresses.find(reader); a != addresses.end()) {
+          std::ostringstream oss;
+          oss << "read";
+          oss << buttonPin;
+          m.Send(a, oss.str());
+        }
+        // Periodically toggle LED1
+        led1on = !led1on;
+        if (const auto a = addresses.find(writer); a != addresses.end()) {
+          std::ostringstream oss;
+          oss << led1on ? "high " : "low ";
+          oss << led1Pin;
+          m.Send(a, oss.str());
+        }
+        // Schedule message to be sent to self after 1s
+        m.SendAfter(address, "loop", std::chrono::seconds(1));
+      },
+      // On Exit Action
+      []() {},
+      // Receivers
+      std::map<const std::string, Receiver>{
+          {"gpio",
+           [&](const Address &sender, std::istream &args) {
+             int p;
+             args >> p;
+             if (const auto a = addresses.find(writer);
+                 a != addresses.end() && p == buttonPin) {
+               // Switch LED2 based on Button state
+               int state;
+               args >> state;
+               std::ostringstream oss;
+               oss << state ? "high " : "low ";
+               oss << led2Pin;
+               m.Send(a, oss.str());
+             }
+           }},
+          {"loop", [&](const Address &sender,
```

Comment:
To me it's confusing to have a message named "loop" and a state named "loop".

Reply:
Done!